src/core/server/http/security_response_headers_config.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+const strictTransportSecuritySchema = schema.object({
+  // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+  maxAge: schema.duration({ defaultValue: '1Y' }),
+  includeSubDomains: schema.boolean({ defaultValue: false }),
+  preload: schema.boolean({ defaultValue: false }),
+});
+
+export const securityResponseHeadersSchema = schema.object({
+  strictTransportSecurity: schema.oneOf([strictTransportSecuritySchema, schema.literal(null)], {
+    defaultValue: strictTransportSecuritySchema.validate({}),
+  }),
+  xContentTypeOptions: schema.oneOf([schema.literal('nosniff'), schema.literal(null)], {
+    // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options
+    defaultValue: 'nosniff',
+  }),
+  referrerPolicy: schema.oneOf(
+    // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy
+    [
+      schema.literal('no-referrer'),
+      schema.literal('no-referrer-when-downgrade'),
+      schema.literal('origin'),
+      schema.literal('origin-when-cross-origin'),
+      schema.literal('same-origin'),
+      schema.literal('strict-origin'),
+      schema.literal('strict-origin-when-cross-origin'),
+      schema.literal('unsafe-url'),
+      schema.literal(null),
+    ],
+    { defaultValue: 'no-referrer-when-downgrade' }
+  ),
+  permissionsPolicy: schema.oneOf([schema.string(), schema.literal(null)], {
+    defaultValue: 'camera=(), microphone=()', // disables camera and microphone access in Kibana and any embedded pages
+  }),
```

Comment:
By default, `Permissions-Policy` is enabled with a value of "camera=(), microphone=()". There is not a lot of authoritative info on this header yet ([the header was previously `Feature-Policy` but has been slightly reworked and renamed](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy)). I figured we can set it with some safe feature policies as a starting point.

Note that `Permissions-Policy` is _subtractive_, so if we do not specify a directive, that is not affected. Kibana shouldn't ever need access to a user's camera or microphone, so what we are saying here is that Kibana (and any embedded pages) are **not allowed** to access these features.

See also: https://github.com/w3c/webappsec-permissions-policy/blob/main/permissions-policy-explainer.md

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy#example leads me to believe that we should be using `Permissions-Policy: camera 'none'; microphone 'none'`. It also looks like this isn't very widely supported per https://caniuse.com/permissions-policy. Perhaps we should just hold off on this one until it matures? Does the lack of this header get flagged by scanners right now?

Reply:
The page is a bit misleading / outdated. That's the format for the old `Feature-Policy` header, it hasn't been updated to the `Permissions-Policy` format.

At any rate based on https://github.com/elastic/kibana/pull/97158#discussion_r614306674 below, I think we will leave this header disabled by default and mark the config prop as experimental, if that sounds good to you.

Reply:
Sounds good to me! Thanks.